cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -846,6 +838,35 @@ private predicate nodeCandFwd2ReadTaint(Content f, Node node, boolean fromArg, C
   )
 }
 
+private predicate readCandFwd2(Content f, Configuration config) {
+  exists(Node node |
+    nodeCandFwd2Read(f, node, _, config) or
+    nodeCandFwd2ReadTaint(f, node, _, config)
+  |
+    nodeCandFwd2(node, _, _, config)
+  )
+}
+
+private predicate readStoreCandFwd2(Content f, Configuration config) {
+  readCandFwd2(f, config) and
+  storeCandFwd2(f, config)
+}
+
+private predicate summaryFwd2(Summary s, Configuration config) {
+  s = TSummaryTaint()
```

Comment:
Why this case? `argumentFlowsThroughFwd2` is only called with `TSummaryReadTaint(f)` and `TSummaryTaintStore(f)`.

Reply:
Fixed.